### DISCUSSION TITLE
Object Array Overwriting itself

### DISCUSSION THREAD

**Discussion by KaneyCain** [react-hook-form/react-hook-form#8504] [Q&A]
Hi can anyone think of a solution to stop these fields inside of an array getting overwritten on seperate pages 

Multi Page Array 
[https://codesandbox.io/s/react-hook-form-wizard-form-forked-w8kk81?file=/src/Step2.js](url)

**Comment by bluebill1049** [comment]
maybe you should store data in a state library? : https://react-hook-form.com/advanced-usage#WizardFormFunnel

**Reply by KaneyCain** [reply]
The example uses little-state-machine to store the form object between pages

**Reply by KaneyCain** [reply]
I think the issue is surrounding the update action overwriting the whole array on each page instead of appending the new values into the objects

**Reply by bluebill1049** [reply]
it's your implementation detail on how data gets insert or replaced.

**Reply by KaneyCain** [reply]
@bluebill1049 So I need to alter the update action to merge the new values into the global state or would this cause issues with registered and unregistered inputs on different pages?
